client/my-sites/plugins/marketplace/index.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * External dependencies
+ */
+import page from 'page';
+
+/**
+ * Internal dependencies
+ */
+import { navigation, siteSelection } from 'calypso/my-sites/controller';
+import config from '@automattic/calypso-config';
+import {
+	renderDomainsPage,
+	renderMarketplacePlugin,
+	renderMarketplaceTestPage,
+	renderMarketplaceThankYou,
+	renderPluginsSetupStatusPage,
+} from './controller';
+import { makeLayout, render as clientRender } from 'calypso/controller';
+
+export default function () {
+	if ( config.isEnabled( 'marketplace-yoast' ) ) {
+		page(
+			'/marketplace/test/:site',
```

Comment:
Many site-related sections include the `site` as an optional argument (meaning, changing `:site` to `:site?`) but then include the `siteSelection` middleware there so they'd ensure that the subsequent middleware will have a site available. We've already included site selection, so we only need to make the site argument optional in order to capture any requests that don't include a site in the route.